src/circuit/gadget/utilities/lookup_range_check.rs

```diff
@@ -0,0 +1,297 @@
+                config: Self::Config,
+            ) -> Result<(), Error> {
+                let mut layouter = SingleChipLayouter::new(cs)?;
+
+                // Load table_idx
+                config.load(&mut layouter)?;
+
+                let num_words = 6;
+                let words_and_expected_final_zs = [
+                    (F::from_u64((1 << (num_words * K)) - 1), F::zero()), // a word that is within num_words * K bits long
+                    (F::from_u64(1 << (num_words * K)), F::one()), // a word that is just over num_words * K bits long
+                ];
+
+                for (words, expected_final_z) in words_and_expected_final_zs.iter() {
+                    let expected_zs = expected_zs::<F, K>(*words, num_words);
+
+                    // Load the value to be decomposed into the circuit.
+                    let words = self.load_private(
+                        layouter.namespace(|| "words"),
+                        config.running_sum,
+                        Some(*words),
+                    )?;
+
+                    let zs = layouter.assign_region(
+                        || "word within range",
+                        |mut region| {
+                            for idx in 0..num_words {
+                                // Assign fixed column to activate lookup.
+                                region.assign_fixed(
+                                    || format!("lookup on row {}", idx),
+                                    config.q_lookup,
+                                    idx,
+                                    || Ok(F::one()),
+                                )?;
+                            }
+
+                            config.lookup_range_check(&mut region, 0, words, num_words)
+                        },
+                    )?;
+
+                    assert_eq!(expected_zs[expected_zs.len() - 1], *expected_final_z);
+
+                    for (expected_z, z) in expected_zs.into_iter().zip(zs.iter()) {
+                        if let Some(z) = z.value() {
+                            assert_eq!(expected_z, z);
+                        }
+                    }
+                }
+
+                Ok(())
+            }
+        }
+
+        {
+            let circuit: MyCircuit<pallas::Base> = MyCircuit {
+                _marker: PhantomData,
+            };
+            let prover = MockProver::<pallas::Base>::run(11, &circuit, vec![]).unwrap();
+            assert_eq!(prover.verify(), Ok(()));
+        }
+    }
+
+    #[cfg(test)]
+    fn expected_zs<F: FieldExt + PrimeFieldBits, const K: usize>(
+        words: F,
+        num_words: usize,
+    ) -> Vec<F> {
+        let chunks = {
+            words
+                .to_le_bits()
+                .iter()
+                .by_val()
+                .take(num_words * K)
+                .collect::<Vec<_>>()
+                .chunks_exact(K)
+                .collect::<Vec<_>>()
+                .iter()
```

Comment:
I think this intermediate collection is unnecessary, as `chunks_exact` returns an iterator of slices that can be passed to `lebs2ip`:
```suggestion
```